### ISSUE TITLE
CMakeLists.txt does not honour CMAKE_INSTALL_PREFIX

### ISSUE THREAD

**Issue by zerodefect** [microsoft/mimalloc#167]
As part of my project, I want to use `ExternalProject_add()` and specify the install location. So in my project, I do something like (untested):

```
################################################################
# Pull source and build
ExternalProject_Add(
    msft_mimalloc
    GIT_REPOSITORY https://github.com/microsoft/mimalloc.git
    GIT_TAG 114d05cd76cdf6b959e8a9ff842d0f0cf20a6071# Release 1.1.0
    TIMEOUT 60
    CMAKE_ARGS -DCMAKE_BUILD_TYPE=RELWITHDEBINFO -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION}
)

#### Copy from install prefix to final location
#### THIS FAILS because INSTALL_PREFIX is not honoured above.
ExternalProject_Add_Step(
    msft_mimalloc CopyToBin
    COMMAND ${CMAKE_COMMAND} -E copy ${EXTERNAL_INSTALL_LOCATION}/lib/libmimalloc-relwithdebinfo.so ${CMAKE_BINARY_DIR}/bin
    DEPENDEES install
)
```

The problem is that the project does not seem to honour $CMAKE_INSTALL_PREFIX. I'm not a CMake expert but the other 10+ CMake project that I build in source work as I expect.

Environment:
- Linux Ubuntu 18.04.3
- CMake v15
- gcc v8.3.0

**Comment by devnexen** [comment]
Little suggestion, what happens if you change the main CMakeLists.txt, the line from
`set(mi_install_dir "lib/mimalloc-${mi_version}")`

to

`set(mi_install_dir "${CMAKE_INSTALL_PREFIX}/lib/mimalloc-${mi_version}")`

?

**Comment by zerodefect** [comment]
Fixed. Opened pull request. Please note that change is untested on anything other than Linux. I do not have access to a Windows dev environment.

**Comment by zerodefect** [comment]
On another note, it would be nice to do away with the `-relwithdebinfo` part of `libmimalloc-relwithdebinfo.so` filename.

**Comment by daanx** [comment]
Now the name of mimalloc shows the difference in builds, like `mimalloc`, `mimalloc-debug` or `mimalloc-secure`. I guess we could use `mimalloc` too for `relwithdebinfo` as it should not matter ... ?

**Comment by zerodefect** [comment]
Good question. My current install step for mimalloc looks like:

```
ExternalProject_Add_Step(
    msft_mm CopyToBin
    COMMAND ${CMAKE_COMMAND} -E copy ${EXTERNAL_INSTALL_LOCATION}/lib/mimalloc-1.1/libmimalloc-relwithdebinfo.so ${CMAKE_BINARY_DIR}/bin/libmimalloc.so
    DEPENDEES install
)
```

...but I would ordinarily expect something like:
```
ExternalProject_Add_Step(
    msft_mm CopyToBin
    COMMAND ${CMAKE_COMMAND} -E copy ${EXTERNAL_INSTALL_LOCATION}/lib/libmimalloc.so ${CMAKE_BINARY_DIR}/bin/libmimalloc.so
    DEPENDEES install
)
```

which would mean doing away with the folder name `mimalloc-1.1` and the build-type version of `libmimalloc-relwithdebinfo.so` so just `libmimalloc.so`

**Comment by zerodefect** [comment]
Any further thoughts?

**Comment by daanx** [comment]
Ah I see. I think leaving out the `relwithdebinfo` from the mimalloc name is ok, but I like to keep the version in the folder  -- both to make it apparent but also to allow  multiple versions to coexist.

**Comment by zerodefect** [comment]
Should I open a pull request?

**Comment by daanx** [comment]
Sure -- thanks!  Hmmm, I just looked in the CMakeFiles and it seems to already do this??
See <https://github.com/microsoft/mimalloc/blob/e946d56ba0c1aa795ffb00f862fd32046a62da8e/CMakeLists.txt#L105>

**Comment by zerodefect** [comment]
Ah yes, that has caught me out before. The problem is that `CMAKE_BUILD_TYPE` must be treated in a case-insensitive way. I'm using CMAKE_BUILD_TYPE=RELWITHDEBINFO (as you can see in the original post above).

I'll open up a PR.

**Comment by daanx** [comment]
Fixed (#173 )
